service/controller/v11/resource/deployment/resource.go

```diff
@@ -126,8 +126,9 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	_, err = deploymentsClient.CreateOrUpdateResponder(res.Response())
+	deploymentExtended, err := deploymentsClient.CreateOrUpdateResponder(res.Response())
 	if err != nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deployment failed", "deployment", deploymentExtended, "error", err)
```

Comment:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment failed; deployment: %#q", deploymentExtended), "stack", microerror.Stack(microerror.Mask(err)))
```

I don't remember if I mentioned it earlier, but we have our own [fmt](https://github.com/giantswarm/fmt) repo with various formatting / style / coding guidelines that is useful to take a look for patterns. Regarding logging, there's also one [page](https://github.com/giantswarm/fmt/blob/master/go/log_messages.md#error-handling).

Note that `LogCtx()` doesn't format its strings/values properly so when dumping variables, it's best to use `fmt.Sprintf()`.